homeassistant/components/recorder/__init__.py

```diff
@@ -1160,6 +1173,37 @@ def _process_one_event(self, event: Event) -> None:
         if not self.commit_interval:
             self._commit_event_session_or_retry()
 
+    def _find_shared_attr_in_db(self, attr_hash: int, shared_attrs: str) -> int | None:
+        """Find shared attributes in the db from the hash and shared_attrs."""
+        #
+        # Avoid the event session being flushed since it will
+        # commit all the pending events and states to the database.
+        #
+        # The lookup has already have checked to see if the data is cached
+        # or going to be written in the next commit so there is no
+        # need to flush before checking the database.
+        #
+        assert self.event_session is not None
```

Comment:
I ended up using the bakery here because sqlalchemy couldn't get the transparent caching to match no matter what I did

<img width="331" alt="Screen Shot 2022-04-25 at 22 56 48" src="https://user-images.githubusercontent.com/663432/165262902-d365ce0e-478b-452f-a9b9-6971aec33bd6.png">

With the bakery its a lot less over head since its not having to construct the query text every time
<img width="183" alt="Screen Shot 2022-04-25 at 22 56 22" src="https://user-images.githubusercontent.com/663432/165262922-a551f248-819d-4352-b0e8-745c3d1aa46a.png">